pyproject.toml

```diff
@@ -1,4 +1,125 @@
+[tool.poetry]
+name = "dowhy"
+
+#
+# 0.0.0 is standard placeholder for poetry-dynamic-versioning
+# any changes to this should not be checked in
+#
+version = "0.0.0"
+description = "DoWhy is a Python library for causal inference that supports explicit modeling and testing of causal assumptions"
+authors = [
+    "PyWhy Community <amshar@microsoft.com>"
+]
+maintainers = []
+license = "MIT"
+documentation = "https://py-why.github.io/dowhy"
+repository = "https://github.com/pywhy/dowhy"
+classifiers = [  
+    'Development Status :: 4 - Beta',
+    'License :: OSI Approved :: MIT License',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+]
+keywords = [
+    'causality',
+    'machine-learning',
+    'causal-inference',
+    'statistics',
+    'graphical-model'
+]
+include = ['docs', 'tests', 'CONTRIBUTING.md']
+readme = 'README.rst'
+
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+
+[tool.poetry-dynamic-versioning.substitution]
+files = ["dowhy/__init__.py"]
+
+#
+# Dependency compatibility notes:
+# * xgboost requires Python >=3.7,<3.11
+# * pygraphviz requires Python >=3.8
+# * networkx requires Python >= 3.8
+# * llvmlite requires Python >=3.6,<3.11
+#
+[tool.poetry.dependencies]
+python = ">=3.8,<3.10"
+scipy = "^1.8.1"
+statsmodels = "^0.13.2"
+numpy = "^1.23.1"
+pandas = "^1.4.3"
+networkx = "^2.8.5"
+sympy = "^1.10.1"
+scikit-learn = "1.0.2"
+pydot = "^1.4.2"
+joblib = "^1.1.0"
+# causalml is wired to use llvmlite 0.36
+llvmlite = "^0.36.0"
+pygraphviz = {version = "^1.9", optional = true}
+matplotlib = {version = "^3.5.2", optional = true}
+# TODO: resolve circular dependency w/ econml
+econml = {version = "*", optional = true}
+tqdm = "^4.64.0"
+
+[tool.poetry.extras]
+plotting = ["matplotlib"]
+
+[tool.poetry.dev-dependencies]
+poethepoet = "^0.16.0"
+flake8 = "^4.0.1"
+black = "^22.6.0"
+isort = "^5.10.1"
+pytest = "^7.1.2"
+twine = "^4.0.1"
+nbformat = "^5.4.0"
+jupyter = "^1.0.0"
+flaky = "^3.7.0"
+causalml = "^0.12.3"
+tensorflow = "^2.9.1"
+keras = "^2.9.0"
+xgboost = "^1.6.1"
+tqdm = "^4.64.0"
+mypy = "^0.971"
+econml = "*"
+pygraphviz = "^1.9"
+rpy2 = {version = "^3.5.2", optional = true}
+
+[build-system]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
+build-backend = "poetry.core.masonry.api"
+
 [tool.pytest.ini_options]
 markers = [
     "advanced: not be to run each time. only on package updates."
 ]
+
+[tool.poe.tasks]
+# stop the build if there are Python syntax errors or undefined names
+_flake8Errors = "flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics"
+_flake8Warnings = "flake8 . --count --exit-zero --statistics"
+_black = 'black .'
+_isort = 'isort .'
+
+# testing tasks
+test = "pytest -v -m 'not advanced' --durations=0 --durations-min=60.0"
+test_focused = "pytest -v -m 'focused'"
+
+jupyter = "jupyter nbconvert --to html docs/source/example_notebooks/dowhy_simple_example.ipynb"
```

Comment:
curious: why do we need to convert this notebook to html?

Reply:
Honestly I just moved this script over from the makefile. I'm not sure if it was part of your workflow for some reason

Reply:
haha, I also don't remember now why it was there. I guess we can safely remove it. Does not appear to do anything critical.